### DISCUSSION TITLE
NetBeans UI refinement and legacy codes clean up

### DISCUSSION THREAD

**Discussion by hantsy** [apache/netbeans#6878] [Ideas]
Not sure NetBeans IDE has a plan for rebuilding the UI and make it more friendly for Windows users, I know it is great on MacOS, but not everyone has MacOS/Macbook.

Like the `Fleet` - a reborn work of Intellij IDEA, is it possible to rebuild NetBeans UI using Java FX and adopt the modern ui layout like Fleet, of course step by step. 

My first NetBeans IDE is 3.4, I have used it to build projects for years. I loved this IDE. After Java EE 7, I moved to IDEA for daily work.

It seems a lot of old features are NOT aligned to the current mainstream in these years. 

For example, the framework in Java EE projects, still includes Struts 1.x, Hibernate, Spring MVC, etc.

I would like use a **Create Spring Boot project** wizard instead, call the rest API via https://start.spring.io

Similarly, we can support Quakrus, Micronaut, Vertx, Helidon in the modern way.

And these projects has it own way to run/debug application, not to deploy to Servlet container or application server.

And the PHP frameworks, a lot of items is out of date, but still there.

And JS has similar issues.

And if possible migrate all language/source file supports to Language Server Protocol, thus we can reuse Lsp implementations to save parser work, https://microsoft.github.io/language-server-protocol/implementors/servers/

**Comment by troizet** [comment]
> It seems a lot of old features are NOT aligned to the current mainstream in these years.

I see the problem here as a lack of hands. Most NetBeans developers are volunteers who work on ide in their free time. And there are not as many of them as I would like. There are no more than 10 regular contributors to NetBeans. Compare the number of contributors in NetBeans with the number of contributors in [Intellij IDEA Communtiy](https://github.com/JetBrains/intellij-community).
For example, all PHP support in NetBeans is handled by one person, @Junichi11, for which I thank him very much!
I think you can try to get involved in NetBeans development and do your best to improve ide.
